File: Project_python/pao_ying_chub.py
#import random
import random as rd

#Greeting
print("Welcome to Pao-Ying-Chub Game!!!")
print("What is your name?")
player_name = input("Enter your name: ")
print(f"Hello! {player_name} nice to meet you")
print("How to play?")
print("R:Rock")
print("S:Scissors")
print("P:Paper")

#set variables
bot_set = ["R","S","P"]
count = 0
win = 0
lose = 0
draw = 0

while True:
    user_act = input("Enter R or S or P :")
    user_act = user_act.upper()
    bot_act = rd.choice(bot_set)
    print(f"You select {user_act},Bot select {bot_act}")

#loop
    if user_act == bot_act:
        print("It's a draw!!!")
        draw +=1
    elif user_act == "R" and bot_act =="P":
        print("You lose!!!")
        lose +=1
    elif user_act == "R" and bot_act == "S":
        print("You win!!!")
        win +=1
    elif user_act == "S" and bot_act =="R":
        print("You lose!!!")
        lose +=1
    elif user_act == "S" and bot_act == "P":
        print("You win!!!")
        win +=1
    elif user_act == "P" and bot_act =="S":
        print("You lose!!!")
        lose +=1
    elif user_act == "P" and bot_act == "R":
        print("You win!!!")
        win +=1
    else:
        print('Error, Plase try again!')
    print("********************")

    print("Play again? (Y/N)")
    next_action = input().upper()
    if next_action == "N":
        #Summary
        print(f"You win:",win, "times")
        print(f"You lose:",lose, "times")
        print(f"You draw:", draw, "times")
        print("See you again next time.")
        break

File: Project_R/pao_ying_chub.r
#Greeting
print("Welcome to Pao-Ying-Chub Game!!!")
print("What's your name?")
name <- readLines("stdin", n=1)
print(paste("Hello!",name," nice to meet you"))
print("How to play?")
print("R:Rock")
print("S:Scissors")
print("P:Paper")

# set data
bot_act <- sample(1:3, 1)
win <- 0
lose <- 0
draw <- 0

#play game
while (TRUE) {
  choose <- ("Enter your choice :
              1 rock, 
              2 paper, 
              3 scissors 
              or 0 for exit: ")
  cat(choose)
  user_act <- as.numeric(readLines("stdin", n = 1))
  if (user_act == 1 || user_act == 2 || user_act == 3 ){
      if(user_act == bot_act){
        draw <- draw + 1
        print("It's a draw!!!")
        print("--------------")
      } else if (user_act == 1 && bot_act == 2){
        lose <- lose + 1
        print("It's a lose!!!")
        print("--------------")
      } else if (user_act == 1 && bot_act == 3){
        win <- win + 1
        print("It's a win!!!")
        print("--------------")
      } else if (user_act == 2 && bot_act == 3){
        lose <- lose + 1
        print("It's a lose!!!")
        print("--------------")
      } else if (user_act == 2 && bot_act == 1){
        win <- win + 1
        print("It's a win!!!")
        print("--------------")
      } else if (user_act == 3 && bot_act == 1){
        lose <- lose + 1
        print("It's a lose!!!")
        print("--------------")
      } else if (user_act == 3 && bot_act == 2){
        win <- win + 1
        print("It's a win!!!")
        print("--------------")
      } else {
        print('Error, Plase try again!')
        print("********************")
      }
  } else if (user_act == 0) {
    print("--------------exit--------------")
    break 
  } else {
    print("Invalid choice. Please try again.")
  } 
}

print(paste("You win:",win,"times"))
print(paste("You lose:",lose,"times"))
print(paste("You draw:",draw,"times"))
print(paste("See you again next time."))
